[UFC-1051] Add docs for react client setting (#7212)
mendix/docs: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md

These settings influence the behavior of the Runtime when running your application.

### 3.1 Use React Client (Beta)

This setting enables the new React client. This client was released into Beta with Mendix 10.7. There are three options:

> - _No_: Don't use the React (default).

Reviewer: Meaning we use the Dojo client.

Also 
```suggestion
- _No_: Don't use the React client (default).
```

> - _Yes_: Use the React client. In this mode you will get consistency errors for imcompatable widgets.

Reviewer 2: ```suggestion
- _Yes_: Use the React client. In this mode, you will get consistency errors for incompatible widgets.
```

> - _Migration mode_: Use the React client, but ignore incompatable widgets. Recommended for trying out the new client.

Reviewer 2: ```suggestion
- _Migration mode_: Use the React client and ignore incompatible widgets. Recommended for trying out the new client.
```

> - _Migration mode_: Use the React client and ignore incompatible widgets. Recommended for trying out the new client.

Reviewer: Should we mention that we will display placeholders for incompatible widgets in this case?

Reviewer 3: Yes, I will add.

Reviewer 3: Looks good?
